File: reservite.api/src/main/java/com/enp/reservite/api/controller/BookingController.java
package com.enp.reservite.api.controller;

/*
 * @(#)BookingController.java 1.0 12/09/2024
 * 
 * El código implementado en este formulario esta protegido
 * bajo las leyes internacionales del Derecho de Autor, sin embargo
 * se entrega bajo las condiciones de la General Public License (GNU GPLv3)
 * descrita en https://www.gnu.org/licenses/gpl-3.0.html
 */

/**
 * Clase controller para gestion de reservas
 *
 * @author eliezer.navarro
 * @version 1.0 | 12/09/2024
 * @since 1.0
 */

import java.util.ArrayList;
import java.util.Date;
import java.util.List;

import org.slf4j.Logger;
import org.slf4j.LoggerFactory;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.http.HttpStatus;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.PostMapping;
import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

import com.enp.reservite.api.entity.Booking;
import com.enp.reservite.api.entity.ErrorDetails;
import com.enp.reservite.api.service.BookingService;

@RestController
@RequestMapping("/api/v1/reservite")
public class BookingController {
	
	private static final Logger logger = LoggerFactory.getLogger(BookingController.class);
	
	@Autowired
	BookingService bookingService;
	
	@GetMapping("/booking")
	public ResponseEntity<?> findBookings(){
		List<Booking> lista = new ArrayList<Booking>();
		try{
			bookingService.findAll().forEach(lista::add);
			if(lista.isEmpty()) {
				ErrorDetails err = new ErrorDetails(new Date(),HttpStatus.OK.toString(),"NO CONTENT");
				return new ResponseEntity<ErrorDetails>(err,HttpStatus.OK);
			}
			return new ResponseEntity<List<Booking>>(lista, HttpStatus.OK);
		}catch(Exception e){
			ErrorDetails err = new ErrorDetails(new Date(),HttpStatus.INTERNAL_SERVER_ERROR.toString(),"INTERNAL SERVER ERROR");
			return new ResponseEntity<ErrorDetails>(err, HttpStatus.INTERNAL_SERVER_ERROR);
		}
	}
	
	@PostMapping("/booking")
	public ResponseEntity<?> createBooking(@RequestBody Booking booking){
		Booking savedBooking;
		try{
			savedBooking = bookingService.save(booking);
			if(savedBooking == null) {
				ErrorDetails err = new ErrorDetails(new Date(),HttpStatus.NOT_FOUND.toString(),"Booking <"+booking+"> not created.");
				return new ResponseEntity<ErrorDetails>(err,HttpStatus.NOT_FOUND);
			}
			return new ResponseEntity<Booking>(savedBooking, HttpStatus.CREATED);
		}catch(Exception e){
			ErrorDetails err = new ErrorDetails(new Date(),HttpStatus.INTERNAL_SERVER_ERROR.toString(), e.getMessage());
			return new ResponseEntity<ErrorDetails>(err, HttpStatus.INTERNAL_SERVER_ERROR);
		}
	}

}

File: reservite.client/app/controllers/BookingController.php
<?php

/* PENDIENTR DE PRUEBAS
 * @(#)BookingController.php 1.0 15/09/2024
 * 
 * El código implementado en este formulario esta protegido
 * bajo las leyes internacionales del Derecho de Autor, sin embargo
 * se entrega bajo las condiciones de la General Public License (GNU GPLv3)
 * descrita en https://www.gnu.org/licenses/gpl-3.0.html
 */

/**
 * Clase Controller para gestion de reservas
 *
 * @author eliezer.navarro
 * @version 1.0 | 15/09/2024
 * @since 1.0
 */

    class BookingController extends Controller{

        public function index(){
            $this->view('index');
        }

        public function reserve(){
            if($_SERVER['REQUEST_METHOD'] == 'POST'){
                $roomType = $_POST['roomType'];
                $checkInDate = $_POST['checkInDate'];
                $checkOutDate = $_POST['checkOutDate'];

                $reservation = $this->model('Booking');
                $response = $reservation->reserveRoom($roomType,$checkInDate,$checkOutDate,$_SESSION['authToken']);

                if($response && $response['status'] == 200){
                    $this->view('confirmation',['booking' => $response['data']]);
                }else{
                    $this->view('booking', ['error' => 'Reservation Failed']);
                }
            }else{
                $this->view('booking');
            }
        }

    }

?>